src/EditorFeatures/Core.Wpf/InlineParamNameHints/InlineParamNameHintsTagger.cs

```diff
@@ -0,0 +1,57 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.VisualStudio.Text;
+using Microsoft.VisualStudio.Text.Editor;
+using Microsoft.VisualStudio.Text.Tagging;
+
+namespace Microsoft.CodeAnalysis.Editor.InlineParamNameHints
+{
+    /// <summary>
+    /// The purpose of this tagger is to convert the InlineParamNameHintsDataTags to
+    /// the InlineParamNameHintsTag, which actually creates the UIElement. It reacts to
+    /// tags changing and updates the adornments accordingly.
+    /// </summary>
+    class InlineParamNameHintsTagger : ITagger<IntraTextAdornmentTag>
+    {
+        private readonly ITagAggregator<InlineParamNameHintDataTag> _tagAggregator;
+        private readonly ITextBuffer _buffer;
+
+        public InlineParamNameHintsTagger(ITextBuffer buffer, ITagAggregator<InlineParamNameHintDataTag> tagAggregator)
+        {
+            _buffer = buffer;
+            _tagAggregator = tagAggregator;
+            _tagAggregator.TagsChanged += _tagAggregator_TagsChanged;
+        }
+
+        private void _tagAggregator_TagsChanged(object sender, TagsChangedEventArgs e)
+        {
+            var spans = e.Span.GetSpans(_buffer);
+            foreach (var span in spans)
+            {
+                TagsChanged?.Invoke(this, new SnapshotSpanEventArgs(new SnapshotSpan(span.Snapshot, 0, span.Snapshot.Length)));
+            }
+        }
+
+        public event EventHandler<SnapshotSpanEventArgs> TagsChanged;
+
+        public IEnumerable<ITagSpan<IntraTextAdornmentTag>> GetTags(NormalizedSnapshotSpanCollection spans)
+        {
+            var tagsList = new List<TagSpan<IntraTextAdornmentTag>>();
+            var dataTags = _tagAggregator.GetTags(spans);
+            foreach(var tag in dataTags)
+            {
+                var dataTagSpans = tag.Span.GetSpans(spans[0].Snapshot);
+                var textTag = tag.Tag;
+                if (dataTagSpans.Count > 0)
+                {
+                    var dataTagSpan = dataTagSpans[0];
```

Comment:
why are we only handlingt eh first of these?

Reply:
I believe it is because we don't really see a feasible way that there are going to be more than one span for the projection buffer since it's unclear how things can be split into multiple spans for this feature. So really we only think dataTagSpans's count is going to be 1, but if it isn't then we're only going to handle the first one.

Reply:
in a case like that, i would only check for a Count of 1 and not  have us do anything for the unexpected multi-span case.

Reply:
So the only time we're ever going to get more than one of these is in some funky projection buffer case where the underlying span or position maps into multiple points. Given we're only caring about the start point we only really ever need to look at the first span if the projection buffer did split. So we're somewhat split between:

1. If we did care about the projection case, grabbing the first span would give us the leftmost position which would be the start of the argument...maybe?
2. We just don't care and can filter out the multi-span case.

Reply:
Changed it so it only handles it if the count is 1.